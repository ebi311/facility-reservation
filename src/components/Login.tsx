import Button from '@material-ui/core/Button';
import React, { useCallback } from 'react';
import styled from 'styled-components';
import { auth } from '../auth';

const Container = styled.div`
  width: 100vw;
  height: 100vh;
  display: flex;
  align-items: center;
  justify-content: center;
`;

export const Login: React.FC = () => {
  const onGoogleAuthClick = useCallback(() => {
    auth().then(() => window.location.reload());
  }, []);
  return (
    <Container>
      <Button variant="outlined" onClick={onGoogleAuthClick}>
        Google でログインする
      </Button>
    </Container>
  );
};
