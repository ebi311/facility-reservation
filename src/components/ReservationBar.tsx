import React, { useCallback, useMemo } from 'react';
import { IReservation } from '../models/IReservation';
import { Property } from 'csstype';
import { makeStyles, Theme } from '@material-ui/core/styles';
import { useHistory } from 'react-router-dom';

type PropsType = {
  reservation: IReservation;
  leftOffset: number;
  beginHour: number;
  hourWidth: number;
  backgroundColor: Property.BackgroundColor;
};

type StyleType = {
  width: number;
  left: number;
  backgroundColor: Property.BackgroundColor;
};

const useStyles = makeStyles<Theme, StyleType>(() => ({
  root: {
    height: '100%',
    position: 'absolute',
    display: 'flex',
    alignItems: 'center',
    width: (p) => p.width + 'px',
    left: (p) => p.left + 'px',
  },
  bar: {
    height: '50%',
    width: '100%',
    backgroundColor: (p) => p.backgroundColor,
    cursor: 'pointer',
  },
}));

export const ReservationBar: React.FC<PropsType> = (props) => {
  const {
    leftOffset,
    reservation,
    hourWidth,
    beginHour,
    backgroundColor,
  } = props;
  const { startDate, endDate } = reservation;

  const width = useMemo(() => {
    const hours = endDate.diff(startDate, 'minute') / 60;
    return hourWidth * hours;
  }, [startDate, endDate, hourWidth]);

  const left = useMemo(() => {
    const beginDate = startDate.set('hour', beginHour).startOf('hour');
    const diffStart = startDate.diff(beginDate, 'minute') / 60;
    return leftOffset + diffStart * hourWidth;
  }, [beginHour, hourWidth, leftOffset, startDate]);

  const style = useStyles({
    width,
    left,
    backgroundColor,
  });

  const history = useHistory();
  const click = useCallback(() => {
    history.push('/reservation/' + reservation.id);
  }, [reservation.id]);

  return (
    <div className={style.root}>
      <div className={style.bar} onClick={click}></div>
    </div>
  );
};
