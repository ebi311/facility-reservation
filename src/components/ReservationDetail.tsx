import React, { useEffect, useMemo } from 'react';
import { useSelector, useDispatch, useStore } from 'react-redux';
import IState from '../status/IState';
import IReservationDetailPage from '../status/IReservationDetailPage';
import { RouteComponentProps } from 'react-router';
import { loadTask } from '../actions/reservationDetailActions';

type Props = RouteComponentProps<{ id: string }>;

const ReservationDetail: React.FC<Props> = props => {
  const taskPage = useSelector<IState, IReservationDetailPage>(
    s => s.reservation,
  );
  const dispatch = useDispatch();
  useEffect(() => {
    loadTask(props.match.params.id, dispatch);
  }, []);
  const loading = useMemo(() => {
    return taskPage.loading ? <div>loading...</div> : null;
  }, [taskPage.loading]);
  return (
    <>
      <div>task page</div>
      {loading}
      <h1>{taskPage.reservation.subject}</h1>
    </>
  );
};

export default ReservationDetail;
