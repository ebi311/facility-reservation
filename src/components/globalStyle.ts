import { createGlobalStyle } from 'styled-components';

const GlobalStyle = createGlobalStyle`
  body, html {
    background-color: #eee;
  }
`;

export default GlobalStyle;
