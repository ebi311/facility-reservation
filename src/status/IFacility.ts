import ISystem from './ISystem';

export default interface IFacility {
  id: string;
  name: string;
  description: string;
  system: ISystem;
}
