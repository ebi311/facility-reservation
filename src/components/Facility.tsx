import Avatar from '@material-ui/core/Avatar/Avatar';
import Button from '@material-ui/core/Button/Button';
import Chip from '@material-ui/core/Chip/Chip';
import Container from '@material-ui/core/Container/Container';
import Grid from '@material-ui/core/Grid/Grid';
import InputLabel from '@material-ui/core/InputLabel/InputLabel';
import Paper from '@material-ui/core/Paper/Paper';
import makeStyles from '@material-ui/core/styles/makeStyles';
import TextField from '@material-ui/core/TextField/TextField';
import DeleteIcon from '@material-ui/icons/Delete';
import DoneIcon from '@material-ui/icons/Done';
import dayjs from 'dayjs';
import React, { useCallback, useEffect, useState } from 'react';
import { Controller, useForm } from 'react-hook-form';
import { useHistory, useParams } from 'react-router-dom';
import { getCurrentUser } from '../auth';
import {
  getFacility,
  postFacility,
  putFacility,
} from '../controllers/facilityController';
import { IFacility } from '../models/IFacility';

const initFacility = (): IFacility => {
  const user = getCurrentUser();
  return {
    id: '',
    name: '',
    note: '',
    system: {
      createDate: new Date(),
      createUser: {
        displayName: user?.displayName || '',
        email: user?.email || '',
        face: user?.photoURL || '',
      },
      lastUpdateUser: {
        displayName: user?.displayName || '',
        email: user?.email || '',
        face: user?.photoURL || '',
      },
      lastUpdate: new Date(),
    },
  };
};

const useStyle = makeStyles((theme) => ({
  root: {
    '& .MuiTextField-root': {
      margin: theme.spacing(1),
    },
  },
  paper: {
    padding: theme.spacing(1),
  },
  rightActions: {
    textAlign: 'right',
  },
  cancelButton: {
    color: theme.palette.error.main,
  },
}));

export const Facility: React.FC = () => {
  const style = useStyle();
  const { id } = useParams<{ id: string }>();
  const [facility, setFacility] = useState(initFacility());
  const { system } = facility;
  const { errors, control, reset, trigger, getValues } = useForm({
    defaultValues: facility,
    mode: 'onBlur',
  });
  useEffect(() => {
    if (!id) return;
    getFacility(id).then((result) => {
      setFacility(result);
      reset(result);
    });
  }, [id]);

  const history = useHistory();
  const onSave = useCallback(async () => {
    const result = await trigger();
    if (!result) return;
    const inputValue = { ...facility, ...getValues() };
    if (!id) {
      // 新規作成
      const id = await postFacility(inputValue);
      history.replace('/facility/' + id);
    } else {
      // 更新
      await putFacility(inputValue);
      window.location.reload();
    }
  }, [id, facility, trigger, getValues]);

  return (
    <Container maxWidth="sm" className={style.root}>
      <Paper className={style.paper}>
        <Controller
          control={control}
          name="name"
          rules={{ required: true }}
          as={
            <TextField
              label="設備名"
              fullWidth
              error={!!errors.name}
              helperText={errors.name ? '必須です' : ''}
            />
          }
        />
        <Controller
          control={control}
          name="note"
          as={<TextField label="詳細" fullWidth multiline value="" />}
        />
        <InputLabel shrink>登録者</InputLabel>
        <p>
          <Chip
            label={system.createUser.displayName}
            avatar={<Avatar src={system.createUser.face} />}
          />
          {dayjs(system.createDate).format('YYYY-MM-DD HH:mm')}
        </p>
        <InputLabel shrink>更新者</InputLabel>
        <p>
          <Chip
            label={system.lastUpdateUser.displayName}
            avatar={<Avatar src={system.lastUpdateUser.face} />}
          />
          {dayjs(system.lastUpdate).format('YYYY-MM-DD HH:mm')}
        </p>
        <Grid container>
          <Grid item xs={6}>
            <Button className={style.cancelButton} startIcon={<DeleteIcon />}>
              削除
            </Button>
          </Grid>
          <Grid item xs={6} className={style.rightActions}>
            <Button
              variant="contained"
              color="primary"
              startIcon={<DoneIcon />}
              onClick={onSave}
            >
              保存
            </Button>
          </Grid>
        </Grid>
      </Paper>
    </Container>
  );
};
