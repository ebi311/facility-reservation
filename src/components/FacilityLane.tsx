import { makeStyles, Theme } from '@material-ui/core/styles';
import { Property } from 'csstype';
import { Dayjs } from 'dayjs';
import React, { useMemo } from 'react';
import { Link } from 'react-router-dom';
import { IFacility } from '../models/IFacility';
import { IReservation } from '../models/IReservation';
import { ReservationBar } from './ReservationBar';

type Props = JSX.IntrinsicElements['div'] & {
  facility: IFacility;
  cellWidth: number;
  backgroundColor: Property.BackgroundColor;
  reservations: IReservation[];
  date?: Dayjs;
};

const useStyles = makeStyles<
  Theme,
  {
    backgroundColor: Property.BackgroundColor;
  }
>((theme) => ({
  header: {
    backgroundColor: (p) => p.backgroundColor,
    '& a': {
      color: (p) => theme.palette.getContrastText(p.backgroundColor),
    },
  },
}));

export const FacilityLane: React.FC<Props> = (props) => {
  const {
    cellWidth,
    facility,
    reservations,
    backgroundColor,
    date,
    ...rootAttr
  } = props;
  const styles = useStyles({ backgroundColor });
  const cells = useMemo(() => {
    const r: JSX.Element[] = [];
    for (let i = 0; i <= 11; i++) {
      const toDate = date?.hour(i + 8).startOf('hour');
      r.push(
        <Link
          key={i}
          className="timeCell"
          to={`/reservation/?date=${toDate?.toISOString() || ''}`}
        ></Link>,
      );
    }
    return r;
  }, []);
  const bars = useMemo(() => {
    return reservations.map((r) => {
      return (
        <ReservationBar
          key={r.id}
          backgroundColor={backgroundColor}
          beginHour={8}
          reservation={r}
          hourWidth={cellWidth}
          leftOffset={100}
        />
      );
    });
  }, [reservations, backgroundColor, cellWidth]);
  return (
    <div {...rootAttr}>
      {bars}
      <div className={`laneHeader ${styles.header}`}>
        <Link to={'/facility/' + facility.id}>{facility.name}</Link>
      </div>
      {cells}
    </div>
  );
};
