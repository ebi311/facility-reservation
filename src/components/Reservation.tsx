import Avatar from '@material-ui/core/Avatar';
import Button from '@material-ui/core/Button';
import Chip from '@material-ui/core/Chip';
import Container from '@material-ui/core/Container';
import FormControl from '@material-ui/core/FormControl';
import Grid from '@material-ui/core/Grid';
import InputLabel from '@material-ui/core/InputLabel';
import MenuItem from '@material-ui/core/MenuItem';
import Paper from '@material-ui/core/Paper';
import Select from '@material-ui/core/Select';
import { makeStyles } from '@material-ui/core/styles';
import TextField from '@material-ui/core/TextField';
import DeleteIcon from '@material-ui/icons/Delete';
import DoneIcon from '@material-ui/icons/Done';
import { DateTimePicker } from '@material-ui/pickers';
import dayjs, { Dayjs } from 'dayjs';
import React, { useCallback, useEffect, useMemo, useState } from 'react';
import { Controller, useForm } from 'react-hook-form';
import { useHistory, useParams } from 'react-router-dom';
import { getFacilities } from '../controllers/facilityController';
import {
  getReservation,
  postReservation,
  putReservation,
} from '../controllers/reservationController';
import { IFacility } from '../models/IFacility';
import { IReservation } from '../models/IReservation';
import { getCurrentUser } from '../auth';

const initReservation = (day: Dayjs): IReservation => {
  const user = getCurrentUser();
  return {
    id: '',
    facilityId: '',
    subject: '',
    description: '',
    startDate: day.startOf('hour'),
    endDate: day.add(1, 'hour').startOf('hour'),
    system: {
      createDate: new Date(),
      createUser: {
        displayName: user?.displayName || '',
        email: user?.email || '',
        face: user?.photoURL || '',
      },
      lastUpdateUser: {
        displayName: user?.displayName || '',
        email: user?.email || '',
        face: user?.photoURL || '',
      },
      lastUpdate: new Date(),
    },
  };
};

const useStyle = makeStyles((theme) => ({
  paper: {
    padding: theme.spacing(1),
    '& > div': {
      marginBottom: theme.spacing(2),
    },
  },
  rightActions: {
    textAlign: 'right',
  },
  cancelButton: {
    color: theme.palette.error.main,
  },
}));

export const Reservation: React.FC = () => {
  const style = useStyle();
  const hour = useMemo(() => {
    const params = new URLSearchParams(window.location.search);
    return dayjs(params.get('date') || undefined);
  }, [window.location.search]);
  const { id } = useParams<{ id: string }>();
  const [reservation, setReservation] = useState(initReservation(hour));
  const [facilities, setFacilities] = useState<IFacility[]>([]);
  const { system } = reservation;
  const { errors, control, reset, trigger, getValues } = useForm<IReservation>({
    defaultValues: reservation,
    mode: 'onBlur',
  });
  useEffect(() => {
    getFacilities().then((result) => {
      setFacilities(result);
    });
    if (!id) return;
    getReservation(id).then((result) => {
      setReservation(result);
      reset(result);
    });
  }, [id]);
  const facilityMenuItems = useMemo(() => {
    return facilities.map((f) => (
      <MenuItem key={f.id} value={f.id}>
        {f.name}
      </MenuItem>
    ));
  }, [facilities]);

  const history = useHistory();
  const onSave = useCallback(async () => {
    const result = await trigger();
    if (!result) return;
    const inputValue = { ...reservation, ...getValues() };
    if (!id) {
      // 新規作成
      const id = await postReservation(inputValue);
      history.replace('/reservation/' + id);
    } else {
      // 更新
      await putReservation(inputValue);
      window.location.reload();
    }
  }, [id, reservation, trigger, getValues]);
  return (
    <Container maxWidth="sm">
      <Paper className={style.paper}>
        <FormControl>
          <InputLabel id="facility-label">設備</InputLabel>
          <Controller
            name="facilityId"
            control={control}
            rules={{ required: true }}
            render={({ value, onChange }) => (
              <Select
                labelId="facility-label"
                value={value}
                onChange={onChange}
              >
                {facilityMenuItems}
              </Select>
            )}
          />
        </FormControl>
        <div style={{ display: 'flex' }}>
          <Controller
            control={control}
            name="startDate"
            render={(data) => {
              return (
                <DateTimePicker
                  value={data.value}
                  onChange={data.onChange}
                  onBlur={data.onBlur}
                  label="開始日時"
                  format="YYYY/MM/DD HH:mm"
                  ampm={false}
                  minutesStep={15}
                />
              );
            }}
          />
          <p>～</p>
          <Controller
            control={control}
            name="endDate"
            render={(data) => {
              return (
                <DateTimePicker
                  value={data.value}
                  onChange={data.onChange}
                  onBlur={data.onBlur}
                  label="終了日時"
                  format="YYYY/MM/DD HH:mm"
                  ampm={false}
                  minutesStep={15}
                />
              );
            }}
          />
        </div>
        <Controller
          control={control}
          name="subject"
          rules={{ required: true }}
          as={
            <TextField
              label="目的"
              fullWidth
              error={!!errors.subject}
              helperText={errors.subject ? '必須です' : ''}
            />
          }
        />
        <Controller
          control={control}
          name="description"
          as={<TextField label="詳細" fullWidth multiline value="" />}
        />
        <InputLabel shrink>登録者</InputLabel>
        <p>
          <Chip
            label={system.createUser.displayName}
            avatar={<Avatar src={system.createUser.face} />}
          />
          {dayjs(system.createDate).format('YYYY-MM-DD HH:mm')}
        </p>
        <InputLabel shrink>更新者</InputLabel>
        <p>
          <Chip
            label={system.lastUpdateUser.displayName}
            avatar={<Avatar src={system.lastUpdateUser.face} />}
          />
          {dayjs(system.lastUpdate).format('YYYY-MM-DD HH:mm')}
        </p>
        <Grid container>
          <Grid item xs={6}>
            <Button className={style.cancelButton} startIcon={<DeleteIcon />}>
              削除
            </Button>
          </Grid>
          <Grid item xs={6} className={style.rightActions}>
            <Button
              variant="contained"
              color="primary"
              startIcon={<DoneIcon />}
              onClick={onSave}
            >
              保存
            </Button>
          </Grid>
        </Grid>
      </Paper>
    </Container>
  );
};
