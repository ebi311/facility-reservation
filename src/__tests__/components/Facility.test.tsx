/**
 * @jest-environment jsdom
 */
/* eslint-disable @typescript-eslint/no-explicit-any */
import { fireEvent, render, waitFor } from '@testing-library/react';
import dayjs from 'dayjs';
import React from 'react';
import { Link, useParams } from 'react-router-dom';
import { Facility } from '../../components/Facility';
import { IFacility } from '../../models/IFacility';
import mockdate from 'mockdate';
import { mocked } from 'ts-jest/utils';
import {
  deleteFacility,
  getFacility,
  postFacility,
  putFacility,
} from '../../controllers/facilityController';

mockdate.set(new Date('2021-07-01T00:00:00'));

jest.mock('../../auth', () => ({
  getCurrentUser: () => ({
    displayName: 'test001',
    email: 'test001@example.com',
    photoURL: 'https://example.com/face/001',
  }),
}));

jest.mock('../../controllers/facilityController');

const history = {
  push: jest.fn(),
  replace: jest.fn(),
};

jest.mock('react-router-dom', () => ({
  useHistory: () => history,
  useParams: jest.fn(() => ({})),
}));

const createFacility = () => ({
  id: 'f001',
  name: '設備001',
  note: '説明001',
  system: {
    createDate: new Date('2021-06-01T00:00:00'),
    createUser: {
      displayName: 'test001',
      email: 'test001@example.com',
    },
    lastUpdateUser: {
      displayName: 'test002',
      email: 'test002@example.com',
    },
    lastUpdate: new Date('2021-06-02T00:00:00'),
  },
});

beforeEach(() => {
  mocked(useParams).mockReturnValue({});
});

describe('初期表示', () => {
  test('新規作成表示', () => {
    // テスト実行
    const { asFragment, getByTestId, queryByTestId } = render(<Facility />);
    // スナップショットの確認
    expect(asFragment()).toMatchSnapshot();
    // 各項目の初期値が表示されているか
    const facilityNameTextbox = getByTestId('facility-name').querySelector(
      'input',
    );
    expect(facilityNameTextbox?.value).toBe('');
    const noteTextbox = getByTestId('note').querySelector('textarea');
    expect(noteTextbox?.value).toBe('');
    const create = getByTestId('create');
    expect(create.textContent).toBe('test0012021-07-01 00:00');
    const update = getByTestId('update');
    expect(update.textContent).toBe('test0012021-07-01 00:00');
    const deleteButton = queryByTestId('delete-button');
    expect(!!deleteButton).toBeFalsy();
  });
  test('既存データ表示', async () => {
    // モック
    mocked(useParams).mockReturnValue({ id: '001' });
    mocked(getFacility).mockResolvedValueOnce(createFacility());
    // テスト実行
    const { getByTestId, queryByTestId } = render(<Facility />);
    await waitFor(() => {
      const facilityNameTextbox = getByTestId('facility-name').querySelector(
        'input',
      );
      return expect(facilityNameTextbox?.value).toBe('設備001');
    });
    const noteTextbox = getByTestId('note').querySelector('textarea');
    expect(noteTextbox?.value).toBe('説明001');
    const create = getByTestId('create');
    expect(create.textContent).toBe('test0012021-06-01 00:00');
    const update = getByTestId('update');
    expect(update.textContent).toBe('test0022021-06-02 00:00');
    const deleteButton = queryByTestId('delete-button');
    expect(!!deleteButton).toBeTruthy();
  });
});

describe('イベント処理', () => {
  test('設備名・説明変更', () => {
    const { getByTestId } = render(<Facility />);
    // イベントを発行するたびに getByXXX で取り直す必要があるので、関数とする
    const facilityNameTextbox = () =>
      getByTestId('facility-name').querySelector('input') as HTMLInputElement;
    const noteTextbox = () =>
      getByTestId('note').querySelector('textarea') as HTMLTextAreaElement;
    // テキストボックスの変更
    fireEvent.change(facilityNameTextbox(), { target: { value: '設備000' } });
    fireEvent.change(noteTextbox(), { target: { value: '説明000' } });
    expect(facilityNameTextbox().value).toBe('設備000');
    expect(noteTextbox().value).toBe('説明000');
  });
  test('削除ボタン クリック', async () => {
    // モック
    mocked(useParams).mockReturnValue({ id: '001' });
    mocked(getFacility).mockResolvedValueOnce(createFacility());
    // テスト実行
    const { getByTestId } = render(<Facility />);
    const deleteButton = getByTestId('delete-button');
    fireEvent.click(deleteButton);
    expect(deleteFacility).toBeCalledWith('001');
    await waitFor(() => expect(history.push).toBeCalledWith('/'));
  });
  test('新規 保存ボタン クリック', async () => {
    // モック
    mocked(postFacility).mockResolvedValueOnce('999');
    const { getByTestId } = render(<Facility />);
    // 保存ボタンクリック
    const saveButton = () => getByTestId('save-button') as HTMLButtonElement;
    fireEvent.click(saveButton());
    // 入力エラーになり、保存処理は実行されない
    expect(postFacility).not.toBeCalled();
    expect(putFacility).not.toBeCalled();
    // 入力して、再度保存する
    const facilityNameTextbox = getByTestId('facility-name').querySelector(
      'input',
    ) as HTMLInputElement;
    const noteTextbox = getByTestId('note').querySelector(
      'textarea',
    ) as HTMLTextAreaElement;
    fireEvent.change(facilityNameTextbox, { target: { value: '設備999' } });
    fireEvent.change(noteTextbox, { target: { value: '説明999' } });
    fireEvent.click(saveButton());
    expect(postFacility).not.toBeCalledWith({
      name: '設備999',
      note: '説明999',
    });
    await waitFor(() =>
      expect(history.replace).toBeCalledWith('/facility/999'),
    );
  });
  test('更新 保存ボタン クリック', async () => {
    // モック
    mocked(useParams).mockReturnValue({ id: '001' });
    mocked(getFacility).mockResolvedValueOnce(createFacility());
    mocked(postFacility).mockResolvedValueOnce('999');
    // テスト実行
    const { getByTestId } = render(<Facility />);
    // 読み込みが終わるまで待つ
    await waitFor(() => {
      const facilityNameTextbox = getByTestId('facility-name').querySelector(
        'input',
      );
      return expect(facilityNameTextbox?.value).toBe('設備001');
    });
    const saveButton = () => getByTestId('save-button') as HTMLButtonElement;
    // 入力して、再度保存する
    const facilityNameTextbox = getByTestId('facility-name').querySelector(
      'input',
    ) as HTMLInputElement;
    const noteTextbox = getByTestId('note').querySelector(
      'textarea',
    ) as HTMLTextAreaElement;
    fireEvent.change(facilityNameTextbox, { target: { value: '設備999' } });
    fireEvent.change(noteTextbox, { target: { value: '説明999' } });
    fireEvent.click(saveButton());
    expect(putFacility).not.toBeCalledWith({
      name: '設備999',
      note: '説明999',
    });
  });
});
