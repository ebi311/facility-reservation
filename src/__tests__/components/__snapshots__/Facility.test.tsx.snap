// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`初期表示 新規作成表示 1`] = `
<DocumentFragment>
  <div
    class="MuiContainer-root makeStyles-root-1 MuiContainer-maxWidthSm"
  >
    <div
      class="MuiPaper-root makeStyles-paper-2 MuiPaper-elevation1 MuiPaper-rounded"
    >
      <div
        class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
        data-testid="facility-name"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
          data-shrink="false"
        >
          設備名
        </label>
        <div
          class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiInput-input"
            name="name"
            type="text"
            value=""
          />
        </div>
      </div>
      <div
        class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
        data-testid="note"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
          data-shrink="false"
        >
          詳細
        </label>
        <div
          class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl MuiInputBase-multiline MuiInput-multiline"
        >
          <textarea
            aria-invalid="false"
            class="MuiInputBase-input MuiInput-input MuiInputBase-inputMultiline MuiInput-inputMultiline"
            name="note"
            rows="1"
            style="height: 0px; overflow: hidden;"
          />
          <textarea
            aria-hidden="true"
            class="MuiInputBase-input MuiInput-input MuiInputBase-inputMultiline MuiInput-inputMultiline"
            readonly=""
            style="visibility: hidden; position: absolute; overflow: hidden; height: 0px; top: 0px; left: 0px; transform: translateZ(0); width: 100%;"
            tabindex="-1"
          />
        </div>
      </div>
      <label
        class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-animated MuiInputLabel-shrink"
        data-shrink="true"
      >
        登録者
      </label>
      <div
        data-testid="create"
      >
        <div
          class="MuiChip-root"
        >
          <div
            class="MuiAvatar-root MuiAvatar-circle MuiChip-avatar"
          >
            <img
              class="MuiAvatar-img"
              src="https://example.com/face/001"
            />
          </div>
          <span
            class="MuiChip-label"
          >
            test001
          </span>
        </div>
        2021-07-01 00:00
      </div>
      <label
        class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-animated MuiInputLabel-shrink"
        data-shrink="true"
      >
        更新者
      </label>
      <div
        data-testid="update"
      >
        <div
          class="MuiChip-root"
        >
          <div
            class="MuiAvatar-root MuiAvatar-circle MuiChip-avatar"
          >
            <img
              class="MuiAvatar-img"
              src="https://example.com/face/001"
            />
          </div>
          <span
            class="MuiChip-label"
          >
            test001
          </span>
        </div>
        2021-07-01 00:00
      </div>
      <div
        class="MuiGrid-root MuiGrid-container"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
        />
        <div
          class="MuiGrid-root makeStyles-rightActions-3 MuiGrid-item MuiGrid-grid-xs-6"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
            data-testid="save-button"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              <span
                class="MuiButton-startIcon MuiButton-iconSizeMedium"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M9 16.2L4.8 12l-1.4 1.4L9 19 21 7l-1.4-1.4L9 16.2z"
                  />
                </svg>
              </span>
              保存
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
